pyproject.toml

```
@@ -1,3 +1,58 @@
+[build-system]
+requires = [
+    "hatchling >= 1.10.0",
+]
+build-backend = "hatchling.build"
+
+[project]
+name = "aiodocker"
+dynamic = ["version"]
+readme = "README.rst"
+description = "A simple Docker HTTP API wrapper written with asyncio and aiohttp."
+license = { text = "Apache 2.0" }
+requires-python = ">=3.8.0"
+
+classifiers = [
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12"
+]
+dependencies = [
+    "aiohttp==3.9.5",
+    "async-timeout==4.0.3",
+    "multidict==6.0.5",
+    "typing_extensions==4.11.0",
+    "yarl==1.9.4"
+]
```

We should keep the original `setup.py`'s *range-based* dependencies, instead of exactly pinning the version here, because `aiodocker` is a library, not a final product package. We must allow our users to choose the exact versions to pin.

```suggestion
dependencies = [
    "aiohttp>=3.8",
    "async-timeout>=4.0.3",
    "typing_extensions>=4.0",
]
```

Also `multidict` and `yarl` are indirect dependencies of aiohttp, we may exclude it from here but keep our desired pinned versions in the "ci" optional dependency set.

We can also remove dependency to `typing_extensions` because the only thing we use from it is `from typing_extensions import Literal` whereas `typing.Literal` is available since Python 3.8.

Additionally, I'd like to keep the trailing comma in multi-line lists or maps if the syntax allows like TOML.

---

I'm going to add an update commit to this PR addressing the above issues.